coding-the-legislation/40_legislation_evolutions.md

```diff
@@ -51,76 +53,111 @@ we get the output:
-    column = FloatCol
-    entity = Person
-    label = u"Former tax replaced by the flat tax on the 1st of Jan 2014"
-    stop_date = date(2013, 12, 31)
-    definition_period = MONTH
+    def formula(self, simulation, period):
+        salary = person('salary', period)
 
-    def function(person, period, legislation):
-        ...
+        return salary * legislation(period).taxes.salary.rate
 ```
 
+If the `flat_tax_on_salary` is calculated for a person **before** the 31st of Dec. 2016 (included), `formula` is used. If it is called **after** the 1st of Jan 2017 (included), `formula_2017` is used.
+
 Note that:
-- A variable can of course have both a `start_date` and a `stop_date`.
-- The `stop_date` is the last day a formula is valid, and not the first day it is not valid anymore.
+- A formula name must always start with `formula`.
+- To define a starting date for a formula, we add to its name a suffix made of an underscore followed by a date.
+  - For instance, `formula_2017_01_01` is active from the 1st of Jan. 2017.
 - When defining a date, the month is given **before** the day.
+- When no month or day is specified, OpenFisca uses '01' as default value.
+  - For instance, `formula_2017` is equivalent to `formula_2017_01_01`.
+- If no date is specified for a formula, OpenFisca will consider that this formula has been active since the dawn of time (or more precisely, since `0001-01-01`, as Python does not handle B.C. dates).
+  - For instance, `formula` is active on `2010`.
+- A formula is active until another formula, starting later, becomes active and replaces it.
+  - For instance, `formula` is active until `2016-12-31` (included). On the day after, `2017-01-01`, `formula_2017` becomes active, and `formula` becomes inactive.
 
-## Formula evolving over time
 
-Some fiscal or benefit mechanism significantly evolve over time, with bigger changes than a simple parameter adjustement.
+## Introduction of a formula
+
+In our previous example, we assumed that `flat_tax_on_salary` had _always_ had a formula, since the dawn of time. This is a reasonable hypothesis if we are only interested in running computations for recent years.
 
-For instance, let's assume that from the 1st of Jan. 2017, our previous `flat_tax_on_salary` will not apply on the first `1000` earned by the person. We can implement this rule with a `DatedVariable`:
+But most fiscal and benefit mechanisms have been introduced at some point. Let's for instance assume that our `flat_tax_on_salary` only appeared in our legislation on the 1st of June 2005. 
+
+This is easily implemented by _dating_ the two formulas:
 
 ```py
-class flat_tax_on_salary(DatedVariable):
+class flat_tax_on_salary(Variable):
     column = FloatCol
     entity = Person
     label = u"Individualized and monthly paid tax on salaries"
     definition_period = MONTH
 
-    @dated_function(start = date(2017, 1, 1))
-    def function_2017(person, period, legislation):
+    def formula_2017(self, simulation, period):
```

Comment:
Same than previous comment: not sure about reordering this chapter, and with the current order, this formula should remain for consistency.